lib/createUris.js

```diff
@@ -0,0 +1,28 @@
+const config = require('../config');
+const { get } = require('lodash');
+const { parse } = require('path');
+
+const createUris = (files, args = {}) => {
+  const createUriOptions = Object.assign(
+    {},
+    config.createUrisDefaultArgs,
+    config.defaultArgs,
+    args
+  );
+
+  return files.map(file => {
+    const fileName = get(file, createUriOptions.fileNamePath);
+    if (!fileName) {
+      throw new Error(`No file name found for file: ${JSON.stringify(file)}`);
+    }
+
+    const parsed = parse(fileName);
+    return {
+      uri: createUriOptions.uri.replace(':documentId', file.id),
+      type: parsed.name.replace(/[^a-z]/gi, ''),
+      fileType: parsed.ext.replace(/[^a-z]/gi, '').toUpperCase()
```

Comment:
The regex used for `type` and `fileType` will not work for any extension with a number (e.g `.mp3`).  Out of curiousty, where will they be used? I'd prefer to just return whatever `ext`/`extname` return because the handle a few funnies instead of rolling out our own regex https://nodejs.org/api/path.html#path_path_extname_path

Reply:
If we _have_ to get rid of the leading `.` then we can perhaps just take that out

Reply:
they are used when displaying file to user e.g. `Document File Name (EXT)`

Reply:
the regex for filetype actually was just to remove `.` but completely see what you mean here, will just have a specific regex for dot and leave rest...

Reply:
Looks good now, can we do the same for `type` please?

Reply:
No, type is working as expected. I want all bet lettered to be stripped from file name.